packages/server/src/login/login-course.service.ts

```diff
@@ -134,4 +136,37 @@ export class LoginCourseService {
         uc.role === previousCourse.role,
     );
   }
+
+  // util functions for converting Khoury Admin data to KOH data
+  private convertKhouryRole(khouryRole: 'TA' | 'Student'): Role {
+    return khouryRole.toLowerCase() === 'ta' ? Role.TA : Role.STUDENT;
+  }
+
+  private parseKhourySemester(khourySemester: string): {
+    season: Season;
+    year: number;
+  } {
+    // parsing time
+    const year = Number(`20${khourySemester.slice(-2)}`);
+    const season = this.parseKhourySeason(
+      khourySemester.slice(0, khourySemester.length - 3),
+    );
+
+    return { season, year };
+  }
+
+  private parseKhourySeason(khourySeason: string): Season {
+    let season = khourySeason;
+    switch (
+      khourySeason // summer sems are the only ones that are diff
+    ) {
+      case 'Summer I':
+        season = 'Summer_1';
+        break;
+      case 'Summer II':
+        season = 'Summer_2';
+        break;
```

Comment:
@jtavera235 just confirming one final time, the summer courses are formatted using Roman numerals like this, right?

Reply:
Yea just wanted to get your eyes on this parsing so we are 100% on the same page

Reply:
So this is on me but turns out the format of the semester is going to be different than originally planned. The admin page is going to return the semester code that appears on Banner. This is a 6 digit string where the first 4 digits are the year and the last 2 digits represent the semester with the semester mappings being:
10 - Fall
30 - Spring
40 _ sum1
50 - SumF
60 - Sum2
However when it is the Fall semester, it maps to the academic year. As an example:
202110 is Fall of 2020 - The academic year is 2021 but the current year of the fall semster is 2020.
202130 is Spring of 2021, etc. The weird rule only applies to Fall Semester